### PR TITLE
rationalize repair type min/max metrics

### PR DESCRIPTION
**Pull request by jbiseda**
#### Problem
Repair slot min/max are collected for each repair type, but min/max across repair types are not uniformly reported.

#### Summary of Changes
- Add metrics for `WindowIndex`, `HighestWindowIndex`, `Orphan` repair request min/max slot values. Only report non-default values.
- Retain `repair-highest-slot` and `repair-orphan` values for continuity with existing metrics. These data points can eventually be removed.
- Remove `repair-total` metric. This metric is the sum of existing `shred-count`, `highest-shred-count`, and `orphan-count` values.

Fixes #

### CODE REVIEW

File: core/src/repair_service.rs
Comment (behzadnouri) [inline core/src/repair_service.rs:489]: Looks like `RepairStatsGroup.min` is always zero.

Comment (jbiseda) [reply core/src/repair_service.rs:493]: Yep, looks that way. Updated.

File: core/src/repair_service.rs
Comment (jbiseda) [reply core/src/repair_service.rs:484]: So this shouldn't be touched? Is there a list somewhere?

File: metrics/src/datapoint.rs
Comment (steviez) [inline metrics/src/datapoint.rs:1]: In general, I like the addition of `Option<_>` as a supported type over sentinel values, but think we could break it into its' own PR ?

Comment (jbiseda) [reply metrics/src/datapoint.rs:1]: I was thinking about separating the `Option<_>` change, but wasn't sure how it would be received without a consumer. I'll pull it out into a a separate PR.

Comment (steviez) [reply metrics/src/datapoint.rs:1]: I think no immediate consumer is fine (hopefully no clippy issue for dead code), especially if you link. This is change is kind of small so feel like I'm nitpicking a little, but in general, like keeping commits separate for separate items (when possible)

File: core/src/repair_service.rs
Comment (steviez) [reply core/src/repair_service.rs:484]: I don't have strong thoughts on this one and can go either way:
- It is arguably redundant to have the total count when we also report the individual counts that comprise the total.
   - Assuming the monitoring Behzad mentioned is a dashboard, the query could be updated to sum the individual fields
- It is also convenient to report a total, less clicks when you're randomly scanning metrics

**Review by steviez — DISMISSED** [review]
I'm good with this as-is. However, let's make sure to get @behzadnouri's sign off / reach consensus in the issue he raised about whether to drop or keep `repair-total`.

**Review by yhchiang-sol — DISMISSED** [review]
LGTM.  Be sure to get @behzadnouri's approval before merging the PR.

File: core/src/repair_service.rs
Comment (jbiseda) [reply core/src/repair_service.rs:484]: @behzadnouri can the metric be updated or should this redundant value be retained? I'm fine either way.

Comment (behzadnouri) [reply core/src/repair_service.rs:484]: this is one of primary metrics we look at during weekly metrics review meeting; so i would rather to keep it.

**Review by steviez — APPROVED** [review]

**Pull request merged**
